src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -824,7 +824,7 @@ def forward(
             attention_masks = attention_masks.to(dtype=dtype)  # fp16 compatibility
             attention_masks = (1.0 - attention_masks) * torch.finfo(dtype).min
 
-        queries = keys = self.with_pos_embed(hidden_states, position_embeddings)
+        queries = keys = self.with_pos_embed(hidden_states, position_embeddings).to(hidden_states.dtype)
```

Comment:
Is this change needed? If `position_embeddings` is created in the correct dtype then we shouldn't need to do this, I think